src/ray/core_worker/transport/direct_actor_task_submitter.cc

```diff
@@ -347,11 +347,18 @@ void CoreWorkerDirectActorTaskSubmitter::FailTaskWithError(
     // Special error for preempted actor. The task "timed out" because the actor may
     // not have sent a notification to the gcs; regardless we already know it's
     // preempted and it's dead.
-    rpc::ActorDeathCause &actor_death_cause = *error_info.mutable_actor_died_error();
-    actor_death_cause.mutable_actor_died_error_context()->set_actor_id(
-        task.task_spec.ActorId().Binary());
-    actor_death_cause.mutable_actor_died_error_context()->set_preempted(
-        task.actor_preempted);
+    auto actor_death_cause = error_info.mutable_actor_died_error();
+    auto actor_died_error_context = actor_death_cause->mutable_actor_died_error_context();
+    actor_died_error_context->set_actor_id(task.task_spec.ActorId().Binary());
+    auto node_death_info = actor_died_error_context->mutable_node_death_info();
+    node_death_info->set_reason(rpc::NodeDeathInfo::AUTOSCALER_DRAIN_PREEMPTED);
+    const std::string reason_message =
+        "the node was inferred to be dead due to draining.";
+    node_death_info->set_reason_message(reason_message);
+    actor_died_error_context->set_error_message(
+        "The actor died because its node was being drained and was unavailable.\n"
+        "\tthe actor's node was preempted: " +
+        reason_message);
```

Comment:
Actually to keep the same behavior as before: let's not set error message for now.

Just set actor id and node death info